### DISCUSSION TITLE
Tkinter in Mojo

### DISCUSSION THREAD

**Discussion by Moosems** [modular/modular#382] [Q&A] [answered]
Does Mojo include or have any GUI libraries?

**Comment by mojodojodev** [comment, marked as answer]
No not yet, but the plan is that you can run Mojo code easily from Python so TKinter will still work, we're not sure how easy this'll be out of the gate, but Chris has said that it was an important part of adoption when Apple transitioned from Objective C to Swift that the two worlds were deeply integrated [on the Lex Fridman podcast](https://www.youtube.com/watch?v=pdJQ8iVTwj8&t=6809s), and that they'll do the same thing with Mojo.

**Reply by cjohnson318** [reply]
FWIW I tried running tkinter from the latest mojo as:

```
from PythonInterface import Python
let tk = Python.import_module("tkinter")

# Create a root window
root = tk.Tk()
```

But I'm getting: Error: "An error occurred in Python." If you've had better luck, I'd be interested to hear how you did it.

**Reply by Moosems** [reply]
Hopefully with the release of the mojo SDK we can test it better because you can't really test GUI's in a Jupyter Notebook of I remember properly.

**Reply by SoleSoul** [reply]
Any news now that the SDK has been released?

**Reply by lattner** [reply]
Modular isn't building any GUI libraries, but we encourage you or other community members to.

**Reply by SoleSoul** [reply]
I was more interested in the question whether we can use TKinter from Mojo or not. I would have tried but it doesn't currently support my OS.

**Reply by Moosems** [reply]
I'm going to try using the Docker route.

**Reply by Moosems** [reply]
Docker refuses to work with me.

**Reply by jackos** [reply]
Hi @SoleSoul yes can confirm that TKinter works both from Linux and upcoming macOS support, haven't tested Windows

**Reply by jackos** [reply]
It works well calling TKinter from Mojo, but using TKinter buttons for example to call Mojo functions you need to package them first, we need to build features to make the developer experience better for that.

**Reply by Moosems** [reply]
@jackos What do you mean by packaging the function?

**Reply by rd4com** [reply]
you can call pyimgui probably

**Reply by ghassenTn** [reply]
from python import Python
var tk = Python.import_module("tkinter")
var root = tk.Tk()
root.mainloop()

this for tkinter gui but there is difficult for calling func with button
